packages/react-big-commerce/spec/Provider.spec.tsx

```diff
@@ -0,0 +1,144 @@
+import type { AnyClient } from "@gadgetinc/api-client-core";
+import { GadgetConnection } from "@gadgetinc/api-client-core";
+import "@testing-library/jest-dom";
+import { render } from "@testing-library/react";
+import React from "react";
+import { mockUrqlClient } from "../../api-client-core/spec/mockUrqlClient.js";
+import { Provider } from "../src/Provider.js";
+
+describe("GadgetProvider", () => {
+  let mockApiClient: AnyClient;
+  const mockBigCommerceInit = jest.fn();
+  let setAuthenticationModeSpy: jest.SpyInstance;
+  const { location } = window;
+
+  beforeAll(() => {
+    // @ts-expect-error mock
+    delete window.location;
+
+    // @ts-expect-error mock
+    window.location = {
+      search: "",
+    };
+
+    (window as any).Bigcommerce = {
+      init: mockBigCommerceInit,
+    };
+  });
+
+  beforeEach(() => {
+    mockApiClient = {
+      connection: new GadgetConnection({
+        endpoint: "https://test-app.gadget.app/endpoint",
+      }),
+    } as any;
+
+    jest.spyOn(mockApiClient.connection, "currentClient", "get").mockReturnValue(mockUrqlClient);
+    setAuthenticationModeSpy = jest.spyOn(mockApiClient.connection, "setAuthenticationMode");
+  });
+
+  afterEach(() => {
+    mockBigCommerceInit.mockReset();
+  });
+
+  afterAll(() => {
+    window.location = location;
+  });
+
+  test("when there is no session parameter", () => {
+    const { container } = render(
+      <Provider api={mockApiClient}>
+        <span>hello world</span>
+      </Provider>
+    );
+
+    expect(setAuthenticationModeSpy).not.toHaveBeenCalled();
+    expect(mockUrqlClient.executeQuery.mock.calls.length).toBe(1);
+    expect(mockUrqlClient.executeQuery.mock.calls[0][0].query.loc.source.body).toMatchInlineSnapshot(`
+      "query BigCommerceSession {
+        currentSession {
+          bigCommerceUserId
+          bigCommerceStore {
+            storeHash
+          }
+          roles {
+            key
+          }
+        }
+      }"
+    `);
+    expect(mockUrqlClient.executeQuery.mock.calls[0][0].variables).toMatchInlineSnapshot(`{}`);
+
+    mockUrqlClient.executeQuery.pushResponse("BigCommerceSession", {
+      data: {
+        currentSession: {
+          bigCommerceUserId: null,
+          bigCommerceStore: null,
+          roles: [
+            {
+              key: "unauthenticated",
+            },
+          ],
+        },
+      },
+      stale: false,
+      hasNext: false,
+    });
+
+    expect(container.outerHTML).toMatchInlineSnapshot(`"<div><span>hello world</span></div>"`);
+    expect(mockBigCommerceInit).not.toHaveBeenCalled();
+  });
+
+  test("when there is a session parameter", () => {
+    window.location.search = "session=test-session-token";
```

Comment:
Thoughts on keeping the query param the same as the one that BigC sends for clarity? `signed_jwt_token` or whatever it is?

Reply:
:+1: